### ISSUE TITLE
shell recorder tests fail

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#1485]
Were these silent errors that were not shown because shell recorder did not report them?

https://build.libelektra.org/job/elektra-incremental/1594

```
109/151 Test #109: testshell_markdown_tutorial_cascading ...***Failed    2.47 sec
protocol file: /tmp/elektraenv.UntnB3qjz
kdb get /sw/tutorial/cascading/#0/current/test

ERROR - STDERR:
“Did not find key”
doesn't match “ Did not find key”

kdb set system/sw/tutorial/cascading/#0/current/test "hello world"
kdb get /sw/tutorial/cascading/#0/current/test
kdb set user/sw/tutorial/cascading/#0/current/test "hello galaxy"
kdb get /sw/tutorial/cascading/#0/current/test
mkdir -p kdbtutorial
cd kdbtutorial
kdb set dir/sw/tutorial/cascading/#0/current/test "hello universe"
kdb get /sw/tutorial/cascading/#0/current/test
cd ..
kdb get /sw/tutorial/cascading/#0/current/test
kdb set system/overrides/test "hello override"
kdb setmeta spec/sw/tutorial/cascading/#0/current/test override/#0 /overrides/test
kdb get /sw/tutorial/cascading/#0/current/test
kdb set /overrides/test "hello user"
kdb get /sw/tutorial/cascading/#0/current/test
kdb rm -r user/sw/tutorial/
kdb rm -r system/sw/tutorial
kdb rm -r system/overrides/test
kdb rm /overrides/test
kdb rm -r spec/sw/tutorial/
shell_recorder /tmp/tmp.7umaCCDRws RESULTS: 32 test(s) done 1 error(s).

        Start 110: testshell_markdown_kdb-complete
110/151 Test #110: testshell_markdown_kdb-complete .........***Failed    1.72 sec
protocol file: /tmp/elektraenv.AmGjgb22h
kdb set /sw/elektra/examples/kdb-complete/level1 foo
kdb set /sw/elektra/examples/kdb-complete/lvl1/lvl2 bar
kdb set /sw/elektra/examples/kdb-complete/lvl1/lvl2/lvl3/lvl4/lvl5 fizz
kdb set /sw/elektra/examples/kdb-complete/buzz fizzBuzz
kdb set /sw/elektra/examples/kdb-complete/#array_1 asdf
kdb set /sw/elektra/examples/kdb-complete/% nothing
kdb complete us --max-depth=1
kdb complete user --max-depth=1
kdb complete user/ --max-depth=1

ERROR - STDOUT:
“user/sw/”
doesn't match “STDOUT-REGEX:”

kdb complete --max-depth=1

ERROR - STDOUT:
“dir
proc
spec
system/
user/”
doesn't match “STDOUT-REGEX:”

kdb complete /sw/elektra/examples/kdb-complete --max-depth=1
kdb complete /sw/elektra/examples/kdb-complete/ --max-depth=1
kdb complete /sw/elektra/examples/kdb-complete/ --min-depth=2 --max-depth=4
kdb rm -r user/sw/elektra/examples/kdb-complete
shell_recorder /tmp/tmp.wrE52TLeXu RESULTS: 21 test(s) done 2 error(s).

The following tests FAILED:
        109 - testshell_markdown_tutorial_cascading (Failed)
        110 - testshell_markdown_kdb-complete (Failed)
```

**Comment by sanssecours** [comment]
> Were these silent errors that were not shown because shell recorder did not report them?

From what I can tell the cause of the first failure is the missing support for leading whitespace in the Markdown Shell Recorder syntax. Commit e8ab5f8b did add support for this feature. The second problem was addressed in commit 832bb7e8. Since both of these tests work fine on my machine: Could you please tell me how I can recreate the test environment of `elektra-incremental`?

**Comment by markus2330** [comment]
Sorry for the noise: the issue only occurred because incremental did not properly purged all cmake files.
